src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -587,8 +592,8 @@ internal override bool NonNullTypes
         {
             get
             {
-                // PROTOTYPE(NullableReferenceTypes): temporary solution to avoid cycle
-                return SyntaxBasedNonNullTypes(((SourceAssemblySymbol)this.ContainingAssembly).GetAttributeDeclarations()) ?? this.UtilizesNullableReferenceTypes;
+                var attributeData = (CommonModuleWellKnownAttributeData)GetAttributesBag().DecodedWellKnownAttributeData;
+                return attributeData?.NonNullTypes ?? UtilizesNullableReferenceTypes;
```

Comment:
>?? [](start = 51, length = 2)

Should this be `&&`?

Reply:
I don't think so. If the module has an attribute, then that attribute wins. If there is no attribute, then we get the value from the options/features that are enabled.